drivers: pwm_nrfx: Add support for center-aligned mode

jimparis opened the pull request:
Add support for up-and-down counter mode, which aligns the center of
each channel's pulses instead of their initial edges.  This is enabled
on a PWM periphral by adding the "center-aligned" property to the
device tree, e.g.:

    &pwm0 {
        status = "okay";
        center-aligned;
        ch0-pin = <15>;
        ch1-pin = <17>;
        ch1-inverted;
    };

Signed-off-by: Jim Paris <jim@jtan.com>

jimparis committed:
drivers: pwm_nrfx: Add support for center-aligned mode

Add support for up-and-down counter mode, which aligns the center of
each channel's pulses instead of their initial edges.  This is enabled
on a PWM periphral by adding the "center-aligned" property to the
device tree, e.g.:

    &pwm0 {
        status = "okay";
        center-aligned;
        ch0-pin = <15>;
        ch1-pin = <17>;
        ch1-inverted;
    };

Signed-off-by: Jim Paris <jim@jtan.com>

anangl commented on drivers/pwm/pwm_nrfx.c:
```diff
@@ -350,6 +359,10 @@ static int pwm_nrfx_pm_control(struct device *dev,
 	 PWM_NRFX_CH_VALUE_INVERTED : PWM_NRFX_CH_VALUE_NORMAL)
 
+#define PWM_NRFX_COUNT_MODE(dev_idx)                                          \
+	(DT_NORDIC_NRF_PWM_PWM_##dev_idx##_CENTER_ALIGNED ?                   \
```
You could use `IS_ENABLED(DT_NORDIC_NRF_PWM_PWM_##dev_idx##_CENTER_ALIGNED)` here. Then, there would be no need for adding these `#ifndef`s below.

jimparis committed:
drivers: pwm_nrfx: use IS_ENABLED to eliminate a bunch of #ifndefs

For the macros that take on 0/1 values, we can use the existing
IS_ENABLED macro to remove a bunch of code.

Signed-off-by: Jim Paris <jim@jtan.com>

jimparis commented:
Rebased on master and added a patch to remove 60 lines of `#ifndef`s based on @anangl 's suggestion
